### PR TITLE
Change flags to flatcar

### PR DESCRIPTION
**Pull request by mazzy89**
Current Mayu still has an old flag for the `autologin` and also some boot flags like `coreos.first_boot=1`

I need this because I've tried to run Flatcar 2247.6.0 (presumably the version corresponding to CoreOS not having any CPU fix) but Mayu has old unsupported flags.

The reason why I need the autologin is because when I start 2247.6.0 on the CP the nodes do not boot and get into emergency state and from VMware console I cannot login so I need to force the login having `autologin` in place.

### CODE REVIEW

**Comment by glitchcrab** [comment]
at the moment Flatcar should still support the Coreos flags (see https://github.com/flatcar-linux/init/pull/17/files), but it makes sense to switch things over.

**Comment by mazzy89** [comment]
True but not true pals 😇

That fix came to Flatcar stable only in one of the latest release to be exact https://www.flatcar-linux.org/releases/#release-2345.3.0

but I'm testing 2247.6.0 Flatcar on puma so here explained.

File: pxemgr/ipxe_handlers.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (mgr *pxeManagerT) ipxeBootScript(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// for ignition we use only 1phase installation without mayu-infopusher
-	kernel := fmt.Sprintf("kernel %s/images/vmlinuz coreos.first_boot=1 initrd=initrd.cpio.gz coreos.config.url=%s?uuid=${uuid}&serial=${serial} systemd.journald.max_level_console=debug verbose log_buf_len=10M "+extraFlags+"\n", mgr.pxeURL(), mgr.ignitionURL())
+	kernel := fmt.Sprintf("kernel %s/images/vmlinuz flatcar.first_boot=1 initrd=initrd.cpio.gz coreos.config.url=%s?uuid=${uuid}&serial=${serial} systemd.journald.max_level_console=debug verbose log_buf_len=10M "+extraFlags+"\n", mgr.pxeURL(), mgr.ignitionURL())
```
Comment (mazzy89) [inline pxemgr/ipxe_handlers.go:50]: Need to check if `coreos.config.url` require change too

Comment (mazzy89) [reply pxemgr/ipxe_handlers.go:50]: seems that the change to support `coreos.config.url` also in Flatcar reached flatcar stable only later than 2247.6.0 see here https://github.com/flatcar-linux/ignition/pull/10

Seems than it is required to change it to `coreos.config.url`

**Comment by mazzy89** [comment]
Tested this on Puma and now the nodes come up correctly with 2247.6.0.

**Comment by mazzy89** [comment]
If for any reasons we wanna keep the backward compatibility with CoreOS then it won't make sense to push this to master. In the other case then we can push safely to the master.

In any case at the moment is fine for me because Mayu on Puma is deployed by this branch and it gives me space and time to test 2247.6.0

File: pxemgr/ipxe_handlers.go (hunk 1 above)
Comment (mazzy89) [reply pxemgr/ipxe_handlers.go]: Changed this.

**Comment by tfussell** [comment]
Should we merge this @mazzy89 ?

**Comment by mazzy89** [comment]
@tfussell no please. Leave it around. Long story short this PR pushes a temp mayu version with the legacy flag for the old Flatcar that we currently are still running.

**Comment by glitchcrab** [comment]
let's get this merged in - we don't need to support coreos anymore as it is long gone. if master has `flatcar` flags then this will work for all flatcar versions.

**Pull request merged**